cranelift-codegen/src/ir/constant.rs

```diff
@@ -0,0 +1,158 @@
+//! Constant representation.
+
+use crate::ir::immediates::Uimm128;
+use core::fmt::{self, Display, Formatter};
+use core::slice::Iter;
+use std::boxed::Box;
+use std::collections::BTreeSet;
+use std::vec::Vec;
+
+/// Symbolizes a constant in the IR that will be stored separately (e.g. in a constant pool)
+///
+/// # Fields:
+///
+/// * data: holds the value of the constant as bytes
+/// * offset: the location at which to store the constant in the constant pool
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd)] // TODO
+pub struct Constant {
+    data: ConstantData,
```

Comment:
I suspect this should be a slice of the original data to avoid copying data around until we actually have to emit code. I'll get to this once we figure out some of the bigger issues.